### PR TITLE
fix(appointments): round up timeslots to next pretty time

### PR DESCRIPTION
**Pull request by miaulalala**

### CODE REVIEW

File: lib/Service/Appointments/AvailabilityGenerator.php [hunk 1]
```diff
@@ -67,7 +67,8 @@ public function generate(AppointmentConfig $config,
 		// E.g. 5m slots should only be available at 10:20 and 10:25, not at 10:17
 		//      when the user opens the page at 10:17.
 		// But only do this when the time isn't already a "pretty" time
-		if ($earliestStart % $config->getIncrement() !== 0) {
+		$prettyTimes = range(0,2700, 300);
```
Comment (ChristophWurst) [inline lib/Service/Appointments/AvailabilityGenerator.php:70]:
```suggestion
		$prettyTimes = range(0, 2700, 300);
```

File: lib/Service/Appointments/AvailabilityGenerator.php [hunk 2]
```diff
@@ -67,7 +67,8 @@ public function generate(AppointmentConfig $config,
...
+		if (!in_array($earliestStart % $config->getIncrement(), $prettyTimes)) {
```
Comment (ChristophWurst) [inline lib/Service/Appointments/AvailabilityGenerator.php:71]:
```suggestion
		if (!in_array($earliestStart % $config->getIncrement(), $prettyTimes, true)) {
```

File: lib/Service/Appointments/AvailabilityGenerator.php [hunk 3]
```diff
@@ -67,7 +67,8 @@ public function generate(AppointmentConfig $config,
 		// E.g. 5m slots should only be available at 10:20 and 10:25, not at 10:17
 		//      when the user opens the page at 10:17.
 		// But only do this when the time isn't already a "pretty" time
-		if ($earliestStart % $config->getIncrement() !== 0) {
+		$prettyTimes = range(0, 2700, 300);
+		if (!in_array($earliestStart % $config->getIncrement(), $prettyTimes, true)) {
```
Comment (ChristophWurst) [inline lib/Service/Appointments/AvailabilityGenerator.php:71]: I'm not sure if there are side effects. The old logic was always specific to the length of the increment with modulo being 0. Example: half hour increments are "beautiful" at :00 and :30.
Now they are "beautiful" at :00, :05, :10, ... :45. Why is it always those 5minute steps regardless of the increment? And what about :50 and :55?

Comment (ChristophWurst) [reply lib/Service/Appointments/AvailabilityGenerator.php:71]: https://3v4l.org/0MXWN

I think only the :00 and :30 should pass the beauty test for a 30m increment

Comment (miaulalala) [reply lib/Service/Appointments/AvailabilityGenerator.php:71]: well the earliest start time for hourly slots with an availability of WED: 10:10 is 10:10. We can either adjust the slot generation to start at 11:00 earliest for full hour slots, 10:30 for half hour slots etc - which is nice - or we can allow a user to start a slot at any 5 minute increments independent of start time - which is also nice.

Forgot to add 50 and 55 to the range :sweat_smile:

Comment (ChristophWurst) [reply lib/Service/Appointments/AvailabilityGenerator.php]: That is not intended. If you book 30min slots at 10:11 the first slot should be 10:30, not 10:15.

Comment (miaulalala) [reply lib/Service/Appointments/AvailabilityGenerator.php]: Ok then we need to modify the slot generation some more to adjust the earlies start time.

Comment (miaulalala) [reply lib/Service/Appointments/AvailabilityGenerator.php]: Slot generation is using the first available timestamp from the JSON to extrapolate all slots. So if you enter 10:10 as your first available timestamp, the date filtering will extrapolate within a whole free day - and do 10:10 to 10:40 etc.

**Review by st3iny — APPROVED** [review]
I tested it and it works as I would expect it to work.

Unit test failures seem to be related though.

**Pull request merged**